codes/c/chapter_divide_and_conquer/hanota.c

```diff
@@ -1,82 +1,81 @@
 /**
  * File: hanota.c
  * Created Time: 2023-10-01
- * Author: Zuoxun (845242523@qq.com)
+ * Author: Zuoxun (845242523@qq.com), lucas(superrat6@gmail.com)
  */
 
 #include "../utils/common.h"
 
+// 假设最多有 1000 个排列
+#define MAX_SIZE 1000
+
 /* 移动一个圆盘 */
-void move(vector *src, vector *tar) {
+void move(int *src, int *srcSize, int *tar, int *tarSize) {
     // 从 src 顶部拿出一个圆盘
-    int *panTemp = vectorBack(src);
-    int *pan = malloc(sizeof(int));
-    *pan = *panTemp;
-    vectorPopback(src);
+    int pan = src[*srcSize - 1];
```

Comment:
Here, the value of the removed plate is set to `0` to avoid subsequent use `memset`
`src[*srcSize - 1] = 0;`